### PR TITLE
Bootstrap Django 4.2

### PR DESCRIPTION
**Pull request by carltongibson**
To be merged after `stable/4.1.x` is branched.

### CODE REVIEW

File: docs/faq/install.txt [hunk 1]
```diff
@@ -54,6 +54,7 @@ Django version Python versions
 3.1            3.6, 3.7, 3.8, 3.9 (added in 3.1.3)
 3.2            3.6, 3.7, 3.8, 3.9, 3.10 (added in 3.2.9)
 4.0, 4.1       3.8, 3.9, 3.10
+4.2            3.9, 3.10, 3.11
```
Comment (carltongibson) [inline docs/faq/install.txt:57]: Brain-burp. We drop versions **after* the LTS not **for** the LTS. 🤦‍♀️

Comment (felixxm) [inline docs/faq/install.txt:57]:
```suggestion
4.2            3.8, 3.9, 3.10, 3.11
```
We should also drop rows for Django 2.2 and 3.1.

File: docs/releases/4.2.txt [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+:ref:`begun the deprecation process for some features
+<deprecated-features-4.2>`.
+
+See the :doc:`/howto/upgrade-version` guide if you're updating an existing
+project.
+
+Python compatibility
+====================
+
+Django 4.2 supports Python 3.9, 3.10, and 3.11. We **highly recommend** and
```
Comment (felixxm) [inline docs/releases/4.2.txt:21]:
```suggestion
Django 4.2 supports Python 3.8, 3.9, 3.10, and 3.11. We **highly recommend** and
```

File: docs/releases/index.txt [hunk 3]
```diff
@@ -20,6 +20,14 @@ versions of the documentation contain the release notes for any later releases.
 
 .. _development_release_notes:
 
+4.1 release
```
Comment (felixxm) [inline docs/releases/index.txt:23]:
```suggestion
4.2 release
```

**Review by felixxm — COMMENTED** [review]
@carltongibson Thanks :+1:

File: docs/faq/install.txt [hunk 4]
```diff
@@ -50,10 +50,9 @@ What Python version can I use with Django?
 ============== ===============
 Django version Python versions
 ============== ===============
-2.2            3.5, 3.6, 3.7, 3.8 (added in 2.2.8), 3.9 (added in 2.2.17)
-3.1            3.6, 3.7, 3.8, 3.9 (added in 3.1.3)
 3.2            3.6, 3.7, 3.8, 3.9, 3.10 (added in 3.2.9)
 4.0, 4.1       3.8, 3.9, 3.10
+4.2            3.8, 3.9, 3.10, 3.11
```
Comment (carltongibson) [inline docs/faq/install.txt:55]: This'll do for now. We can adjust when adding 3.11 support to 4.1.

**Review by felixxm — APPROVED** [review]

**Pull request merged**